libsolidity/Types.h

```diff
@@ -218,6 +218,9 @@ class Type: private boost::noncopyable, public std::enable_shared_from_this<Type
 
 	virtual std::string toString(bool _short) const = 0;
 	std::string toString() const { return toString(false); }
+	/// @returns the canonical name of this type for use in function signatures.
+	/// @param _addDataLocation if true, includes data location for reference types if it is "storage".
+	virtual std::string canonicalName(bool /*_addDataLocation*/) const { return toString(true); }
```

Comment:
The default implementation does not make use of this parameter and one way to silence the compiler warning is to not name the parameter.